gui/elements/_viewer.py

```diff
@@ -306,5 +345,99 @@ def update_statusBar(self):
+        """
+        if event.pos is None:
+            return
+
+        self._update_index(event)
+        if event.is_dragging:
+            if self.annotation and 'Shift' in event.modifiers:
+                if self._active_markers:
+                    layer = self.layers[self._active_markers]
+                    index = layer._selected_markers
+                    if index is None:
+                        pass
+                    else:
+                        layer.data[index] = [self._index[1],self._index[0],*self._index[2:]]
+                        layer._refresh()
+                        self._update_statusBar()
+        else:
+            self._update_active_layers()
+            self._update_statusBar()
+
+    def on_mouse_press(self, event):
+        if event.pos is None:
+            return
+
+        if self.annotation:
+            if self._active_markers:
+                layer = self.layers[self._active_markers]
+                if 'Meta' in event.modifiers:
+                    index = layer._selected_markers
+                    if index is None:
+                        pass
+                    else:
+                        if isinstance(layer.size, (list, ndarray)):
+                            layer._size = delete(layer.size, index)
+                        layer.data = delete(layer.data, index, axis=0)
+                        layer._selected_markers = None
+                        self._update_statusBar()
+                elif 'Shift' in event.modifiers:
+                    pass
+                else:
+                    if isinstance(layer.size, (list, ndarray)):
+                        layer._size = append(layer.size, 10)
+                    coord = [self._index[1],self._index[0],*self._index[2:]]
+                    layer.data = append(layer.data, [coord], axis=0)
+                    layer._selected_markers = len(layer.data)-1
+                    self._update_statusBar()
+
+    def on_key_press(self, event):
+        if event.native.isAutoRepeat():
+            return
+        else:
+            if event.key == ' ':
+                if self.annotation:
+                    self._annotation_history = True
+                    self.layers.viewer._qt.view.interactive = True
+                    self.annotation = False
+                    self._qt.canvas.native.setCursor(QCursor())
+                else:
+                    self._annotation_history = False
+            elif event.key == 'Shift':
+                if self.annotation and self._active_markers:
+                    self._qt.canvas.native.setCursor(Qt.PointingHandCursor)
+            elif event.key == 'Meta':
+                if self.annotation and self._active_markers:
+                    self._qt.canvas.native.setCursor(Qt.ForbiddenCursor)
+            elif event.key == 'a':
+                cb = self.layers._qt.layersControls.annotationCheckBox
+                cb.setChecked(not cb.isChecked())
+
```

Comment:
This is such a hacky way to implement the desired functionality. What about switching these if elseif structures to QActions and setting shortcuts to these actions?

Reply:
sure, happy to change. Right now I'm using vispy to detect the keypress events and there's actually a bug where if focus if shifted to another widget the keypress events won't be detected. clicking back in the viewer restores focus to vispy and they work again. i think @jni noticed this behavior when he was playing around.

@AhmetCanSolak how do you feel about the need to fix this now vs after we merge making a separate PR that cleans up all keyboard shortcuts and adds in some new ones like ctrl+w to close? I definitely don't see what I have here as a long term solution.

Reply:
it is okay to not solve right away IMO, we can let this PR hang here and can merge later after fixing. Or you can comment out this part only and later on these lines will let us to remember the intention and we can merge the PR and have a follow-up PR.

Reply:
@AhmetCanSolak what about this implementation makes you think that it’s hacky?